From ebe09291793adcc1ca5559bf86c50466f6967b15 Mon Sep 17 00:00:00 2001
From: Arpit Agarwal <arp@apache.org>
Date: Thu, 17 Dec 2015 10:36:07 -0800
Subject: [PATCH 1044/2848] HDFS-9565.
 TestDistributedFileSystem.testLocatedFileStatusStorageIdsTypes
 is flaky due to race condition. (Contributed by
 Wei-Chiu Chuang)

(cherry picked from commit 72a3f649e4db2b1cc18a8e203c321cc1832435e4)

Change-Id: I2bfce97829a5c7c38f3c68397cbab5dd61ad1086
---
 .../hadoop/hdfs/TestDistributedFileSystem.java     |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
index b3f5234..bc00a55 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
@@ -710,6 +710,7 @@ public void testLocatedFileStatusStorageIdsTypes() throws Exception {
       final int repl = 2;
       DFSTestUtil.createFile(fs, testFile, blockSize, numBlocks * blockSize,
           blockSize, (short) repl, 0xADDED);
+      DFSTestUtil.waitForReplication(fs, testFile, (short) repl, 30000);
       // Get the listing
       RemoteIterator<LocatedFileStatus> it = fs.listLocatedStatus(testFile);
       assertTrue("Expected file to be present", it.hasNext());
-- 
1.7.9.5

